src/hooks/useGenericHook.ts

```diff
@@ -0,0 +1,43 @@
+import { ChangeEvent, ChangeEventHandler, useState } from "react";
+import { SelectData } from "../utils/data";
+
+//pamietam o regule 1hook = 1czynnosc ale tu jest ok :)
+
+export function useGenericHook(selectData: SelectData) {
+  const [optionData, setOptionData] = useState<SelectData>(selectData);
+  const [showSelectedOptionData, setSelectedShowOptionData] =
+    useState<SelectData>([]);
+  const [optionOnChange, setOptionOnChange] = useState("");
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (optionOnChange === "" || optionData.length === 0) return;
+
+    const newArrayWithSelectedOption = optionData.filter((el) => {
+      return el.option === optionOnChange;
+    });
+
+    const newArrayWithoutSelectedOption = optionData.filter((el) => {
+      return el.option !== optionOnChange;
+    });
```

Comment:
```suggestion
    const { newArrayWithSelectedOption, newArrayWithoutSelectedOption } = optionData.reduce((acc, el) => {
    el.option === optionOnChange ? acc.newArrayWithSelectedOption.push(el) : acc.newArrayWithoutSelectedOption.push(el);
    return acc;
}, { newArrayWithSelectedOption: [], newArrayWithoutSelectedOption: [] });
```

Reply:
to mozna w reactie stosowac pushowanie czy nie?:D

Reply:
reduce zawsze pod koniec zwraca całkowicie nowe obiekty z nowymi referencjami więc w tym przypadku można

Reply:
problem z ts, krzyczy w linii 18 dla el:

Argument of type '{ option: string; }' is not assignable to parameter of type 'never'.ts(2345)

     ? acc.newArrayWithSelectedOption.push(el)
          : acc.newArrayWithoutSelectedOption.push(el);
          
dodalem typy i nie pomoglo:
{
      newArrayWithSelectedOption: SelectData;
      newArrayWithoutSelectedOption: SelectData;
    }

Reply:
No bo chyba powinno być SelectData[]

Reply:
nie pomaga, dalej el podkreslone

const {
      newArrayWithSelectedOption,
      newArrayWithoutSelectedOption,
    }: {
      newArrayWithSelectedOption: { option: string }[];
      newArrayWithoutSelectedOption: { option: string }[];
    } = optionData.reduce(
      (acc, el) => {
        el.option === optionOnChange
          ? acc.newArrayWithSelectedOption.push(el)
          : acc.newArrayWithoutSelectedOption.push(el);
        return acc;
      },
      { newArrayWithSelectedOption: [], newArrayWithoutSelectedOption: [] }
    );